BeerProject/src/test/java/wawer/kamil/beerproject/controllers/UserControllerTest.java

```diff
@@ -1,125 +1,248 @@
-        assertEquals(HttpStatus.OK, controller.findUserByUserId(ID).getStatusCode());
+    @DisplayName("Test - should return response entity with saved user")
+    public void should_return_response_entity_with_saved_user() throws UsernameAlreadyExistsException {
+        //given
+        when(userService.addNewUser(userRequest)).thenAnswer(invocation -> {
+            UserRequest argument = invocation.getArgument(0, UserRequest.class);
+            UserResponse userResponse = new UserResponse();
+            userResponse.setId(2L);
+            userResponse.setUsername(argument.getUsername());
+            userResponse.setPassword(argument.getPassword());
+            userResponse.setEmail(argument.getEmail());
+            userResponse.setGrantedAuthorities(argument.getGrantedAuthorities());
+            userResponse.setEnabled(true);
+            userResponse.setCredentialsNonExpired(true);
+            userResponse.setAccountNonLocked(true);
+            userResponse.setAccountNonExpired(true);
+            return userResponse;
+        });
+
+        //when
+        ResponseEntity<UserResponse> newUserResponseEntity = controller.createNewUser(userRequest);
+
+        //than
+        assertThat(newUserResponseEntity).isNotNull();
+        assertThat(newUserResponseEntity.getStatusCode()).isEqualTo(HttpStatus.CREATED);
+        assertThat(newUserResponseEntity.getBody()).isNotNull();
+        assertThat(newUserResponseEntity.getBody().getId()).isEqualTo(2L);
+        assertThat(newUserResponseEntity.getBody().getUsername()).isEqualTo("user");
+        assertThat(newUserResponseEntity.getBody().getPassword()).isEqualTo("user");
+        assertThat(newUserResponseEntity.getBody().getEmail()).isEqualTo("test@email.com");
+        assertThat(newUserResponseEntity.getBody().isAccountNonExpired()).isTrue();
+        assertThat(newUserResponseEntity.getBody().isAccountNonLocked()).isTrue();
+        assertThat(newUserResponseEntity.getBody().isCredentialsNonExpired()).isTrue();
+        assertThat(newUserResponseEntity.getBody().isEnabled()).isTrue();
     }
 
     @Test
-    public void should_return_saved_user_which_equals_to_user_saved_by_controller(){
-        when(mapper.map(userDTO, User.class)).thenReturn(user);
-        when(service.createNewUser(user)).thenReturn(user);
-        when(mapper.map(user, UserDTO.class)).thenReturn(userDTO);
-        assertEquals(ResponseEntity.status(HttpStatus.CREATED).body(userDTO),controller.createNewUser(userDTO));
+    @DisplayName("Test - should throw exception when username is unavailable during create")
+    public void should_throw_exception_when_username_is_unavailable_during_create() throws UsernameAlreadyExistsException {
+        //given
+        when(userService.addNewUser(userRequest)).thenThrow(UsernameAlreadyExistsException.class);
+
+        //when
+        assertThatThrownBy(() -> controller.createNewUser(userRequest));
     }
```

Comment:
może tam gdzie assert to zrobić //then bo assert to już koniec testu, i zamiast //given to //when

Reply:
jw

Reply:
jw